### PR TITLE
Add tests / tox / travis

### PR DESCRIPTION
**Pull request by yakky**
WiP work to add at least minimal tests and CI

### CODE REVIEW

**Comment by jezdez** [comment]
That's awesome! I would drop Python 2.6 though, as it's really not supported by the Python core team anymore and we shouldn't care about it anymore.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - "if [[ $TRAVIS_PYTHON_VERSION == '3.5' ]]; then export PYVER=py35; fi"
+  - "if [[ ${DJANGO}z != 'z' ]]; then export TOXENV=$PYVER-$DJANGO; fi"
+
+# command to run tests, e.g. python setup.py test
+script: COMMAND='coverage run' tox -e$TOXENV
+
+before_install:
+  - pip install codecov
+after_success:
+  - codecov
```
Comment (jezdez) [inline .travis.yml]: are both services needed?

File: test_helper.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        ('fr', gettext('French')),
+        ('it', gettext('Italiano')),
+    ),
+    FILE_UPLOAD_TEMP_DIR=mkdtemp(),
+    SITE_ID=1,
+)
+
+
+def run():
+    from djangocms_helper import runner
```
Comment (jezdez) [inline test_helper.py]: hmm, I don't think we should depend on 3rd party (despite it being a package maintained by you!) for running simple Django tests, it seems like using `settings.configure` like https://github.com/carljm/django-secure/blob/master/runtests.py would suffice for this

File: .travis.yml (hunk 1 above)
Comment (yakky) [inline .travis.yml]: Obviously not. coveralls removed (but I'm open to any coverage history tool)

File: test_helper.py (hunk 2 above)
Comment (yakky) [inline test_helper.py]: Will check that. Just habit

File: tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# -*- coding: utf-8 -*-
```
Comment (mauricioabreu) [inline tests/__init__.py]: Is this header necessary?
I don't see any characters that could cause encoding errors.

File: tests/base.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        from django.contrib.auth.models import AnonymousUser
+        request = self.request_factory.get(path, secure=secure)
+
+        if not user:
+            user = AnonymousUser()
+        request.user = user
+        request._cached_user = user
+        request.session = {}
+        if secure:
+            request.environ['SERVER_PORT'] = str('443')
```
Comment (mauricioabreu) [inline tests/base.py]: What happens if we do not use `str` here?

**Comment by yakky** [comment]
I think it's a good habit too always add the utf8 encoding header everywhere. It makes code much more consistent

**Comment by yakky** [comment]
Regarding str(): my idea is to add Unicode literals later, str() will force the correct string type

File: tests/base.py (hunk 4 above)
Comment (yakky) [inline tests/base.py]: Regarding str(): my idea is to add Unicode literals later, str() will force the correct string type

File: tests/__init__.py (hunk 3 above)
Comment (yakky) [inline tests/__init__.py]: I think it's a good habit too always add the utf8 encoding header everywhere. It makes code much more consistent

**Comment by yakky** [comment]
This is complete on my side

File: tests/__init__.py (hunk 3 above)
Comment (mauricioabreu) [inline tests/__init__.py]: I got your point.

File: tests/base.py (hunk 4 above)
Comment (mauricioabreu) [inline tests/base.py]: Understood.

**Comment by mauricioabreu** [comment]
LGTM :+1:

**Comment by yakky** [comment]
@jezdez is this ok for you?

**Comment by mauricioabreu** [comment]
It looks like he already accepted your changes.
You can go ahead and merge it. it is a good improvement, why should not we apply this?

Good job! :-)

**Comment by yakky** [comment]
@mauricioabreu thanks :)

**Pull request merged**